heron/common/src/java/com/twitter/heron/common/basics/ByteAmount.java

```diff
@@ -0,0 +1,247 @@
+  private static final long GB = MB * 1024;
+  private static final long MAX_MB = Math.round(Long.MAX_VALUE / MB);
+  private static final long MAX_GB = Math.round(Long.MAX_VALUE / GB);
+
+  public static final ByteAmount ZERO = ByteAmount.fromBytes(0);
+  private final long bytes;
+
+  private ByteAmount(long bytes) {
+    this.bytes = bytes;
+  }
+
+  /**
+   * Creates a ByteAmount value in bytes.
+   *
+   * @param bytes value in bytes to represent
+   * @return a ByteAmount object repressing the number of bytes passed
+   */
+  public static ByteAmount fromBytes(long bytes) {
+    return new ByteAmount(bytes);
+  }
+
+  /**
+   * Creates a ByteAmount value in megabytes. If the megabytes value
+   * is >= Long.MAX_VALUE / 1024 / 1024, the byte representation is capped at Long.MAX_VALUE.
+   *
+   * @param megabytes value in megabytes to represent
+   * @return a ByteAmount object repressing the number of MBs passed
+   */
+  public static ByteAmount fromMegabytes(long megabytes) {
+    if (megabytes >= MAX_MB) {
+      return new ByteAmount(Long.MAX_VALUE);
+    } else {
+      return new ByteAmount(megabytes * MB);
+    }
+  }
+
+  /**
+   * Creates a ByteAmount value in gigabytes. If the gigabytes value
+   * is >= Long.MAX_VALUE / 1024 / 1024 / 1024, the byte representation is capped at Long.MAX_VALUE.
+   *
+   * @param gigabytes value in gigabytes to represent
+   * @return a ByteAmount object repressing the number of GBs passed
+   */
+  public static ByteAmount fromGigabytes(long gigabytes) {
+    if (gigabytes >= MAX_GB) {
+      return new ByteAmount(Long.MAX_VALUE);
+    } else {
+      return new ByteAmount(gigabytes * GB);
+    }
+  }
+
+  /**
+   * Returns the number of bytes associated with the ByteValue.
+   * @return number of bytes
+   */
+  public long asBytes() {
+    return bytes;
+  }
+
+  /**
+   * Converts the number of bytes to megabytes, rounding if there is a remainder. Because of loss
+   * of precision due to rounding, it's strongly advised to only use this method when it is certain
+   * that megabytes were using during the construction of the object via fromMegabytes. Only In that
+   * scenario is there a guarantee that precision will not be lost when calling this method.
+   * @return returns the ByteValue in MBs or 0 if the value is < (1024 * 1024) / 2
+   */
+  public long asMegabytes() {
+    return Math.round((double) bytes / MB);
+  }
+
+  /**
+   * Converts the number of bytes to gigabytes, rounding if there is a remainder. Because of loss
+   * of precision due to rounding, it's strongly advised to only use this method when it is certain
+   * that gigabytes were using during the construction of the object via fromGigabytes. Only In that
+   * scenario is there a guarantee that precision will not be lost when calling this method.
+   * @return returns the ByteValue in GBs or 0 if the value is < (1024 * 1024 * 1024) / 2
+   */
+  public long asGigabytes() {
+    return Math.round((double) bytes / GB);
```

Comment:
Agree. 
We could return float or double from these methods, and let the caller perform context specific transformation?